providers/class.two-factor-email.php

```diff
@@ -120,7 +141,10 @@ public function authentication_page( $user ) {
 			return;
 		}
 
-		$this->generate_and_email_token( $user );
+		if ( ! $this->user_has_token( $user->ID ) ) {
+			$this->generate_and_email_token( $user );
```

Comment:
Email a new code only if none found for the user. Previously we were emailing a new code every time user enters an incorrect code.